kolibri/content/api.py

```diff
@@ -217,6 +220,23 @@ def filter_kind(self, queryset, name, value):
             return queryset.exclude(kind=content_kinds.TOPIC).order_by("lft")
         return queryset.filter(kind=value).order_by("lft")
 
+    def filter_in_lesson(self, queryset, name, value):
+        """
+        Show only content associated with this lesson
+
+        :param queryset: all content nodes
+        :param value: id of target lesson
+        :return: content nodes for this lesson
+        """
+        resources = Lesson.objects.get(id=value).resources
+        contentnode_id_list = [node['contentnode_id'] for node in resources]
+        # adapted from https://codybonney.com/creating-a-queryset-from-a-list-while-preserving-order-using-django/
+        clauses = ' '.join(["WHEN {}.id='{}' THEN {}".format(models.ContentNode._meta.db_table,
+                                                             pk, i) for i, pk in enumerate(contentnode_id_list)])
+        ordering = 'CASE {} END'.format(clauses)
+        return queryset.filter(pk__in=contentnode_id_list) \
+                       .extra(select={'ordering': ordering}, order_by=('ordering',))
```

Comment:
I'm not a django expert, but I thought that constructing raw SQL like this was to be avoided?

Reply:
It is preferable, but this ordering is not possible without it.

The question is whether it is necessary to do this ordering here, or whether it would be easier to delegate it to the frontend.